filters/colorinterp/ColorinterpFilter.cpp

```diff
@@ -0,0 +1,221 @@
+
+#include "color-ramps.hpp"
+
+namespace pdal
+{
+
+static std::vector<std::string> ramps = {"awesome_green", "black_orange", "blue_hue", "blue_red", "heat_map", "pestel_shades", "blue_orange"};
+
+static PluginInfo const s_info = PluginInfo(
+    "filters.colorinterp",
+    "Assigns RGB colors based on a dimension and a ramp",
+    "http://pdal.io/stages/filters.colorinterp.html" );
+
+CREATE_STATIC_PLUGIN(1, 0, ColorinterpFilter, Filter, s_info)
+
+std::string ColorinterpFilter::getName() const { return s_info.name; }
+
+
+#define GETRAMP(name) \
+    if (pdal::Utils::iequals(#name, rampFilename)) \
+    { \
+        VSILFILE* vsifile; \
+        GByte* location(0); \
+        int size (0); \
+        location = name; \
+        size = sizeof(name); \
+        rampFilename = "/vsimem/" + std::string(#name) + ".png"; \
+        vsifile = VSIFileFromMemBuffer(rampFilename.c_str(), location, size, FALSE); \
+    }
+//
+std::shared_ptr<pdal::gdal::Raster> openRamp(std::string& rampFilename)
+{
+    // If the user] selected a default ramp name, it will be opened by
+    // one of these macros if it matches. Otherwise, we just open with the
+    // GDALOpen'able the user gave us
+
+    // GETRAMP will set rampFilename to the /vsimem filename it
+    // used to actually open the file, and from then on it can be treated
+    // like any other GDAL datasource.
+
+    GETRAMP(awesome_green);
+    GETRAMP(black_orange);
+    GETRAMP(blue_hue);
+    GETRAMP(blue_red);
+    GETRAMP(heat_map);
+    GETRAMP(pestel_shades);
+
+    std::shared_ptr<pdal::gdal::Raster> output (new pdal::gdal::Raster(rampFilename.c_str()));
+    return output;
+}
+
+void ColorinterpFilter::addArgs(ProgramArgs& args)
+{
+    std::string dimension;
+    args.add("dimension", "Dimension to interpolate", dimension, "Z");
+    args.add("minimum", "Minimum value to use for scaling", m_min);
+    args.add("maximum", "Maximum value to use for scaling", m_max);
+    args.add("ramp", "GDAL-readable color ramp image to use", m_colorramp, "pestel_shades");
+    args.add("invert", "Invert the ramp direction", m_invertRamp, false);
+    args.add("stddev", "Compute minimum/maximum given this many standard deviations", m_stdDevThreshold, 0.0);
+    m_interpDim = Dimension::id(dimension);
+}
+
+void ColorinterpFilter::addDimensions(PointLayoutPtr layout)
+{
+    layout->registerOrAssignDim("Red",
+        Dimension::defaultType(Dimension::Id::Red));
+    layout->registerOrAssignDim("Green",
+        Dimension::defaultType(Dimension::Id::Green));
+    layout->registerOrAssignDim("Blue",
+        Dimension::defaultType(Dimension::Id::Blue));
+}
+
+void ColorinterpFilter::initialize()
+{
+    gdal::registerDrivers();
+
+    m_raster = openRamp(m_colorramp);
+    m_raster->open();
+
+    log()->get(LogLevel::Debug) << getName() << " raster connection: "
+                                             << m_raster->m_filename << std::endl;
+}
+
+bool ColorinterpFilter::processOne(PointRef& point)
```

Comment:
`processOne()` assumes streaming mode, but you can't use streaming mode with this filter because you have to compute statistics on all the points.  Should probably rename this so that the system doesn't think you CAN stream with it.